### PR TITLE
Drop deprecated `allowPrivilegedContainers` field from the Shoot API

### PR DESCRIPTION
**Pull request by shafeeqes**
**How to categorize this PR?**

/kind cleanup

**What this PR does / why we need it**:
Drop deprecated `allowPrivilegedContainers` field from the Shoot API.
Follow-up of #8989  

**Which issue(s) this PR fixes**:
Part of #8921 

**Special notes for your reviewer**:
/hold until `release-v1.90` branch is created

**Release note**:

```breaking user
Deprecated `.spec.kubernetes.allowPrivilegedContainers` field in the Shoot API is now removed.
```

### CODE REVIEW

**Review by ary1992 — COMMENTED** [review]
Thanks!
/lgtm

**Review by rfranzke — COMMENTED** [review]
/approve

File: plugin/pkg/shoot/validator/admission_test.go [hunk 1]
```diff
@@ -574,9 +573,9 @@ var _ = Describe("validator", func() {
 					shoot.Spec.Maintenance.ConfineSpecUpdateRollout = ptr.To(confine)
 					shoot.Status.LastOperation = operation
 					if specChange {
-						shoot.Spec.Kubernetes.AllowPrivilegedContainers = ptr.To(
-							oldShoot.Spec.Kubernetes.AllowPrivilegedContainers == nil ||
-								!(*oldShoot.Spec.Kubernetes.AllowPrivilegedContainers))
+						shoot.Spec.Kubernetes.KubeControllerManager = &core.KubeControllerManagerConfig{
+							NodeMonitorGracePeriod: &metav1.Duration{Duration: 100 * time.Second},
+						}
```
Comment (ScheererJ) [inline plugin/pkg/shoot/validator/admission_test.go:578]: How does removal of `allowPrivilegedContainers` make new `kube-controller-manager` settings necessary?

**Review by ScheererJ — COMMENTED** [review]
Thanks for the change. I have one question, though.

File: plugin/pkg/shoot/validator/admission_test.go (hunk 1 above)
Comment (shafeeqes) [reply plugin/pkg/shoot/validator/admission_test.go:578]: It could be any spec change. I just chose this.

**Review by ScheererJ — APPROVED** [review]
/lgtm
/approve

**Comment by shafeeqes** [comment]
Force pushed to resolve conflicts.

**Pull request merged**